community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala

```diff
@@ -66,26 +68,63 @@ abstract class CompiledExecutionResult(completion: CompletionListener, statement
     stringWriter.getBuffer.toString
   }
 
-  override def dumpToString(writer: PrintWriter) = formatOutput(statement.readOperations, writer, columns, toList, queryStatistics())
+  override def dumpToString(writer: PrintWriter) = {
+    val builder = Seq.newBuilder[Map[String, String]]
+    doInAccept(populateDumpToStringResults(builder))
+    formatOutput(writer, columns, builder.result(), queryStatistics())
+  }
+
+  /*
+   * NOTE: This should ony be used for testing, it creates an InternalExecutionResult
+   * where you can call both toList and dumpToString
+   */
+  def toEagerIterableResult: InternalExecutionResult = {
+    val dumpToStringBuilder = Seq.newBuilder[Map[String, String]]
+    val resultBuilder = Seq.newBuilder[Map[String, Any]]
+    doInAccept{ (row) =>
+      populateResults(resultBuilder)(row)
+      populateDumpToStringResults(dumpToStringBuilder)(row)
+    }
+    val result = resultBuilder.result()
+    val iterator = result.toIterator
+    new CompiledExecutionResult(taskCloser, statement) {
+      override def executionMode: ExecutionMode = self.executionMode
+
+      override def javaColumns: util.List[String] = self.javaColumns
+
+      override def accept[EX <: Exception](visitor: ResultVisitor[EX]): Unit = throw new UnsupportedOperationException
 
-  override def queryStatistics(): InternalQueryStatistics = InternalQueryStatistics()
+      override def executionPlanDescription(): InternalPlanDescription = self.executionPlanDescription()
+
+      override def toList = result.map(Eagerly.immutableMapValues(_, materialize)).toList
+
+      override def dumpToString(writer: PrintWriter) = formatOutput(writer, columns, dumpToStringBuilder.result(), queryStatistics())
+
+      override def next() = Eagerly.immutableMapValues(iterator.next(), materialize)
+
+      override def hasNext = iterator.hasNext
+    }
+
+  }
+
+  override def queryStatistics() = InternalQueryStatistics()
 
   private var successful = false
-  protected def success(): Unit = {
+  protected def success() = {
     successful = true
   }
 
   override def close(): Unit = {
-    if (innerIterator != null) {
-      innerIterator.close( )
-    }
-    statement.close()
-    completion.complete(success=successful)
+    taskCloser.close(success = successful)
```

Comment:
I was told that in Scala convention, `=`are not used for functions which does not return any value (ie, java `void`). Also, `Unit` is superfluous.